src/policy/policy.cpp

```diff
@@ -237,6 +239,34 @@ bool IsWitnessStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
                     return false;
             }
         }
+
+        // Check MAX_STANDARD_TAPSCRIPT_STACK_ITEM_SIZE policy limit for Taproot spends.
+        if (witnessversion == 1 && witnessprogram.size() == WITNESS_V1_TAPROOT_SIZE && !p2sh) {
+            // Taproot spend (non-P2SH-wrapped, version 1, witness program size 32; see BIP 341)
+            auto stack = MakeSpan(tx.vin[i].scriptWitness.stack);
+            if (stack.size() >= 2 && !stack.back().empty() && stack.back()[0] == ANNEX_TAG) {
+                // Drop annex
```

Comment:
We should probably policy-reject annex here for now?

Reply:
They are policy-invalid, though it's implemented as a script verification flag (`SCRIPT_VERIFY_DISCOURAGE_UNKNOWN_ANNEX`). Any particular reason why you prefer it here?

Reply:
Less expensive/faster to reject?

Reply:
My thinking was initially that there wouldn't be a performance benefit, as taproot execution to the point where the annex is reached is very cheap - but then again, there may be a valid, expensive input, and another one with an annex.

I've moved the annex policy rule to policy/policy.cpp (which also removes a validation flag).

This makes me wonder if we'd want the same with unknown leaf versions, but a reason not to may be that that would mean putting knowledge about defined/valid leaf versions in two places. So I'm going to stick with the rule "implement it where it's simplest", which means annex here, and leaf versions there.